subprojects/gradle/instrumentation-test-impact-analysis/src/main/kotlin/com/avito/instrumentation/impact/TestBytecodeAnalyzeAction.kt

```diff
@@ -303,22 +315,28 @@ abstract class TestBytecodeAnalyzeAction : WorkAction<TestBytecodeAnalyzeAction.
         )
     }
 
-    private fun getRootIdByScreen(
-        config: InstrumentationTestImpactAnalysisExtension,
-        context: Context,
-        targetClasses: Set<JavaClass>
-    ): Map<Screen, RootId> {
-        val idFieldExtractor = IdFieldExtractor.Impl(fieldName = config.screenMarkerMetadataField.get())
-        val screenToIds: Set<IdFieldExtractor.ScreenToId> = idFieldExtractor.extract(
-            context = context,
-            targetClasses = targetClasses
-        )
-        val duplicateCheckResult = DuplicateIdChecker.Impl(config.unknownRootId.get()).check(screenToIds)
-        if (duplicateCheckResult.hasDuplicates) {
-            ciLogger.info("There are duplicate id's: ${duplicateCheckResult.ids}")
-        }
+    private fun getScreenToModule(projectModules: Collection<InternalModule>): Collection<ScreenToModulePath> {
+
+        val sourceSets = projectModules.flatMap { it.androidTestConfiguration.sourceSets() }
 
-        return screenToIds.toMap()
+        ciLogger.info("projectModules=$projectModules")
+
+        val androidModules = projectModules.filter {
+            it.project.pluginManager.hasPlugin("com.android.library") || it.project.pluginManager.hasPlugin("com.android.application")
```

Comment:
```suggestion
            it.project.isAndroid()
```

We have an extension already.

Reply:
thanks, fixed